cachito/workers/pkg_managers/pip.py

```diff
@@ -2119,6 +2119,13 @@ def resolve_pip(path, request, requirement_files=None, build_requirement_files=N
         build_requirement_files = _default_requirement_file_list(path, devel=True)
     else:
         build_requirement_files = _get_absolute_pkg_file_paths(path, build_requirement_files)
+    # Verify each file path exists in the repository
+    for file_path in requirement_files:
```

Comment:
Just a minor nitpick: if you moved this check to inside the `_dowload_from_requirement_files` function, you wouldn't need to duplicate the check.